Add TH-350US version. Fixes #10887 and #11487.

gwholtha1 opened the pull request:
TYT [US software version](https://www.twowayradioforum.com/t/help-programming-a-tyt-350th-analogue-ht/8916/11) sets up the TH-350 for channel names that are 8 characters long. Current CHIRP TH-350 driver expects 5 characters for channel names, which results in garbled names when programming. This update creates a TH-350US option to properly program US-configured radios.

kk7ds commented on chirp/drivers/th350.py:
```diff
@@ -393,3 +393,210 @@ def _get_memobjs(self, number):
+        self._memobj = bitwise.parse(self.MEM_FORMAT, self._mmap)
+
+    # Override uvb5 set_memory to allow for 8 character channel names
+    def set_memory(self, mem):
```
Instead of reproducing `set_memory()` entirely here, can you please modify the one in the base class to use a class variable? Something like:
```
class Th350Radio:
    NAME_LENGTH = 5
    def get_settings(self):
        ...
        rf.name_length = self.NAME_LENGTH
    def set_memory(self, mem):
        ...
        # Use self.NAME_LENGTH instead of hard-coded 5

class Th350USRadio(Th350Radio):
    NAME_LENGTH = 8
    # The rest of your stuff, but exclude set_memory() and get_features()
```

kk7ds commented:
Is there any way to determine the US/non-US model from the clone procedure? If so, It would be best to detect the right model when the user selects TH-350, or at least prevent them from choosing the US model when they have the non-US, etc.

kk7ds commented on chirp/drivers/th350.py:
```diff
@@ -393,3 +393,210 @@ def _get_memobjs(self, number):
+class Th350USRadio(Th350Radio):
+    """TYT TH-350US"""
+    VENDOR = "TYT"
+    MODEL = "TH-350US"
```
Can you leave `MODEL` alone and use `VARIANT = 'US'` please?

gwholtha1 commented:
Thanks for the review. I added a class variable for the channel name length. This required adding the class variable to the uvb5 driver in order to modify the set_memory function, but I don't have that radio to test it.

I also changed the th350 driver to use VARIANT instead of MODEL.

Great suggestion to determine US vs. non-US model from the clone procedure, but I'm not sure how to do that. I checked the identifier string at the start of the radio memory, but it's the same for both variants.

kk7ds commented on chirp/drivers/th350.py:
```diff
@@ -393,3 +395,166 @@ def _get_memobjs(self, number):
+@directory.register
+class Th350USRadio(Th350Radio):
+    """TYT TH-350US"""
+    VENDOR = "TYT"
```
Not necessary to have this here, but won't hurt either.

kk7ds commented:
I just pushed to:

1. Rebase
2. Squash the fixups into the main commit (per the PR guidelines)
3. Fix the bug linkage (Needs to be `Fixes #xxx`, per the PR guidelines) in the commit message
4. Snipped out the extraneous `VENDOR` while I was in there.

Thanks!

gwholtha1 commented:
Thank you!

gwholtha1 commented:
@kk7ds This also fixes #11422 and probably #9343.

kk7ds commented:
Can you comment on those? I'll close as appropriate, thanks!

gwholtha1 commented:
Done. Thank you!
